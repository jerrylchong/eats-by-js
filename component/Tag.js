import React from 'react';
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native';

const Tag = (props) => {
    return (
        <TouchableOpacity>
            <View style = {styles.container}>
                <Text style = {styles.text}>
                    {props.name}
                </Text>
            </View>
        </TouchableOpacity>
    )
}

export default Tag

const styles = StyleSheet.create({
    container: {
        backgroundColor: '#ff6961',
        borderRadius: 10,
        marginRight: 5
    },
    text: {
        fontSize: 10,
        color: 'white',
        paddingHorizontal: 9,
        paddingVertical: 3,
        fontFamily: 'Ubuntu-Bold'
    }
})