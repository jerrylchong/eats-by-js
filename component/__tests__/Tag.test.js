import React from 'react';
import renderer from 'react-test-renderer';
import Tag from '../Tag';
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import {TouchableOpacity} from "react-native";

test('renders correctly', () => {
    const tree = renderer.create(<Tag/>).toJSON();
    expect(tree).toMatchSnapshot();
})

Enzyme.configure({ adapter: new Adapter() });
test('test onPress functionality', () => {
    const onPressEvent = jest.fn();
    onPressEvent.mockReturnValue('onPress invoked');
    const wrapper = shallow(<Tag onPress={onPressEvent}/>);
    wrapper.find(TouchableOpacity).props().onPress();
    expect(onPressEvent.mock.calls.length).toBe(1);
})