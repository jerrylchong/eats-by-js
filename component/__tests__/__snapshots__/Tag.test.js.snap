// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#ff6961",
        "borderRadius": 10,
        "marginRight": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "Ubuntu-Bold",
          "fontSize": 10,
          "paddingHorizontal": 9,
          "paddingVertical": 3,
        }
      }
    />
  </View>
</View>
`;
